js/noms/src/path-test.js

```diff
@@ -277,8 +281,44 @@ suite('Path', () => {
     t('.foo[42]bar', 'Invalid operator: b');
     t('#foo', 'Invalid operator: #');
     t('!foo', 'Invalid operator: !');
-    t('@foo', 'Invalid operator: @');
-    t('@key', 'Invalid operator: @');
+    t('@foo', 'Unsupported annotation: @foo');
+    t('@key', 'Cannot use @key annotation at beginning of path');
     t('.foo[42]@soup', 'Unsupported annotation: @soup');
+    t('.foo@key', 'Cannot use @key annotation on: .foo');
+    t('.foo@key()', '@key annotation does not support arguments');
+    t('.foo@key(42)', '@key annotation does not support arguments');
+    t('.foo@type()', '@type annotation does not support arguments');
+    t('.foo@type(42)', '@type annotation does not support arguments');
+  });
+
+  test('type annotation', async () => {
+    const m = new Map([
```

Comment:
It's used further to test a type annotation, but I've pulled out the key-values to make this nicer.